### PR TITLE
ESP32: Optimize the time taken by esp32 workflows

### PR DESCRIPTION
**Pull request by sweetymhaiske**
#### Problem
* Some jobs in esp32 takes huge time for compilation.
* All clusters app and temperature measurement app was getting build for all sdkconfig*.defaults present.

#### Change overview
* Performed the analysis on the difference between all sdkconfig*.defaults present in all-clusters-app and temperature measurement app and kept the sdkconfig*.defaults which is a superset.
* Added the job for ipv6 only app.

#### Testing
* Successful CI.

### CODE REVIEW

**Comment by dhrishi** [comment]
@sweetymhaiske Please remove `sdkconfig_devkit.defaults` from `all-clusters-app` as well as `temperature-measurement-app` as they are not required now

File: .github/workflows/examples-esp32.yaml [hunk 1]
```diff
@@ -61,53 +61,69 @@ jobs:
...
               timeout-minutes: 5
-              run: scripts/examples/esp_example.sh shell
+              run: scripts/examples/esp_example.sh shell sdkconfig.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/shell/esp32/build/chip-shell.elf \
                      example_binaries/$BUILD_TYPE-build/chip-shell.elf
             - name: Build example Temperature Measurement App
               timeout-minutes: 25
```
Comment (dhrishi) [inline .github/workflows/examples-esp32.yaml:112]: Please reduce the timeout to 5 minutes (just like other examples)

File: .github/workflows/examples-esp32.yaml [hunk 2]
```diff
@@ -61,53 +61,69 @@ jobs:
                    .environment/gn_out/.ninja_log
                    .environment/pigweed-venv/*.log
             - name: Build example All Clusters App
-              timeout-minutes: 25
-              run: scripts/examples/esp_example.sh all-clusters-app
+              timeout-minutes: 10
+              run: scripts/examples/esp_example.sh all-clusters-app sdkconfig_m5stack.defaults
```
Comment (dhrishi) [inline .github/workflows/examples-esp32.yaml:65]: M5Stack has an ESP32 inside it. So there's no separate need to compile sdkconfig.defaults. M5Stack has the extra display component.

So, if the compilation works fine for M5Stack then it should work fine for ESP32 Devkit as well

File: .github/workflows/examples-esp32.yaml [hunk 3]
```diff
@@ -61,53 +61,69 @@ jobs:
...
+              run: scripts/examples/esp_example.sh shell sdkconfig.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/shell/esp32/build/chip-shell.elf \
                      example_binaries/$BUILD_TYPE-build/chip-shell.elf
             - name: Build example Temperature Measurement App
               timeout-minutes: 25
-              run: scripts/examples/esp_example.sh temperature-measurement-app
+              run: scripts/examples/esp_example.sh temperature-measurement-app sdkconfig.optimize.defaults
```
Comment (dhrishi) [inline .github/workflows/examples-esp32.yaml:113]: Similarly, here sdkconfig.optimize.defaults has a couple of extra options as compared to sdkconfig.defaults. So, if this compiles fine, then we can safely assume that the temperature-measurement-app with sdkconfig.defaults will compile fine as well

File: .github/workflows/examples-esp32.yaml [hunk 4]
```diff
@@ -61,53 +61,69 @@ jobs:
...
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/temperature-measurement-app/esp32/build/chip-temperature-measurement-app.elf \
                      example_binaries/$BUILD_TYPE-build/chip-temperature-measurement-app.elf
+            - name: Build example IPV6 Only App
```
Comment (dhrishi) [inline .github/workflows/examples-esp32.yaml:119]: Nit: IPv6

**Comment by sweetymhaiske** [comment]
> @sweetymhaiske Please remove `sdkconfig_devkit.defaults` from `all-clusters-app` as well as `temperature-measurement-app` as they are not required now

Done.

**Comment by andy31415** [comment]
Could you please check that after these changes

```
./scripts/build/build_examples.py --platform esp32 build
```

still works?

I would like to keep it stable until we get to use it in our vscode/CI builds.

**Comment by sweetymhaiske** [comment]
> ./scripts/build/build_examples.py --platform esp32 build
Yes, it works fine.

**Review by dhrishi — APPROVED** [review]

**Comment by dhrishi** [comment]
@andy31415 @bzbarsky-apple PTAL

**Review by mspang — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by dhrishi** [comment]
Can we merge this?

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**